Explorer/Assets/DCL/AvatarRendering/Wearables/Systems/ResolveWearableByPointerSystem.cs

```diff
@@ -183,32 +188,61 @@ private void FinalizeWearableDTO([Data] bool defaultWearablesResolved, in Entity
             {
                 if (!promiseResult.Succeeded)
                 {
+                    //No wearable representation is going to be possible 
                     foreach (string pointerID in promise.LoadingIntention.Pointers)
-                    {
-                        wearableCatalog.TryGetWearable(pointerID, out IWearable component);
-                        SetDefaultWearables(defaultWearablesResolved, component, in bodyShape);
-                        component.IsLoading = false;
-                    }
+                        ReportAndFinalizeWithError(pointerID);
                 }
                 else
                 {
+                    var failedDTOList = WearableComponentsUtils.POINTERS_POOL.Get();
+                    failedDTOList.AddRange(promise.LoadingIntention.Pointers);
+                    
                     foreach (WearableDTO assetEntity in promiseResult.Asset.Value)
                     {
-                        if (!wearableCatalog.TryGetWearable(assetEntity.metadata.id, out IWearable component))
+                        bool isWearableInCatalog = wearableCatalog.TryGetWearable(assetEntity.metadata.id, out var component);
+
+                        try
+                        {
+                            Assert.IsTrue(isWearableInCatalog); //A wearable that has a DTO request should already have an empty representation in the catalog at this point
+                        }
+                        catch (AssertionException e)
                         {
-                            ReportHub.LogError(new ReportData(GetReportCategory()), $"Cannot finalize wearable DTO: {assetEntity.metadata.id}");
+                            ReportHub.LogError(new ReportData(GetReportCategory()), $"Requested wearable {assetEntity.metadata.id} is not in the catalog");
                             continue;
                         }
 
-                        try { component.ResolveDTO(new StreamableLoadingResult<WearableDTO>(assetEntity)); }
-                        catch (AssertionException e) { ReportHub.LogError(new ReportData(GetReportCategory()), $"Cannot apply the DTO to the wearable {component.GetUrn()}: {e.Message}"); }
+                        try
+                        {
+                            component.ResolveDTO(new StreamableLoadingResult<WearableDTO>(assetEntity));
```

Comment:
The same applies here actually (I missed that bit before). If you want to fail gracefully make `ResolveDTO` return `bool` instead and handle it here.